### PR TITLE
Add CI on Windows

### PR DESCRIPTION
**Pull request by IRDonch**
### Summary

Add Windows CI builds and fix/work around any issues that they found.

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
- [ ] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have [linked related issues](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: .github/workflows/pr_checks.yml [hunk 1]
```diff
@@ -6,13 +6,22 @@ on:
...
-  build_and_tests_on_python:
+  build_and_test:
     strategy:
       fail-fast: false
       matrix:
+        os: ['ubuntu-20.04', 'windows-2016']
         python-version: ['3.6', '3.7', '3.8', '3.9']
-    runs-on: ubuntu-latest
+        exclude:
+        # pycocotools-windows doesn't have a wheel for Python 3.9
```
Comment (zhiltsov-max) [inline .github/workflows/pr_checks.yml:20]: Why do we need a wheel? We should be able to build from sources with Cython. 3.9 looks like a thing we need to support (for example, ubuntu 21.04 already uses it).

File: datumaro/plugins/icdar_format/extractor.py [hunk 2]
```diff
@@ -89,8 +89,10 @@ def _load_localization_items(self):
 
         image_dir = osp.join(self._path, IcdarPath.IMAGES_DIR)
         if osp.isdir(image_dir):
-            images = { osp.splitext(osp.relpath(p, image_dir))[0]: p
-                for p in find_images(image_dir, recursive=True) }
+            images = {
+                osp.splitext(osp.relpath(p, image_dir))[0].replace('\\', '/'): p
```
Comment (zhiltsov-max) [inline datumaro/plugins/icdar_format/extractor.py:93]: Maybe it's time to update `find_images()` itself? It has the same pattern of use everywhere.

File: datumaro/util/image.py [hunk 3]
```diff
@@ -78,18 +81,15 @@ def save_image(path, image, create_dir=False, dtype=np.uint8, **kwargs):
...
 
-        params = []
+        ext = osp.splitext(path)[1]
+        image_bytes = encode_image(image, ext, dtype=dtype, **kwargs)
 
-        ext = path[-4:]
-        if ext.upper() == '.JPG':
-            params = [
-                int(cv2.IMWRITE_JPEG_QUALITY), kwargs.get('jpeg_quality', 75)
-            ]
```
Comment (zhiltsov-max) [inline datumaro/util/image.py:89]: Why this code was removed? OpenCV and PIL have different jpeg quality defaults.

File: datumaro/plugins/icdar_format/extractor.py (hunk 2 above)
Comment (IRDonch) [reply datumaro/plugins/icdar_format/extractor.py:93]: Not everywhere. Many formats don't build a lookup table from the `find_images` results, and the path separator doesn't become an issue for them.

File: datumaro/util/image.py (hunk 3 above)
Comment (IRDonch) [reply datumaro/util/image.py:89]: This code is reused through `encode_image`.

**Comment by IRDonch** [comment]
I just realized I forgot to add a changelog entry. WIll do so shortly.

File: CHANGELOG.md [hunk 4]
```diff
@@ -17,6 +17,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Datumaro no longer depends on scikit-image
   (<https://github.com/openvinotoolkit/datumaro/pull/379>)
 - `Dataset` remembers export options on saving / exporting for the first time (<https://github.com/openvinotoolkit/datumaro/pull/386>)
+- When using the OpenCV backend, Datumaro now saves images
```
Comment (zhiltsov-max) [inline CHANGELOG.md:20]: Probably, lines in changelog should be terser.

Comment (IRDonch) [reply CHANGELOG.md:20]: Well... any suggestions for how to improve it? It's as terse as I could make it while also communicating all relevant information.

Comment (zhiltsov-max) [reply CHANGELOG.md:20]: It is a great explanation for the PR description field, but not a changelog entry.

File: CHANGELOG.md [hunk 5]
```diff
@@ -17,6 +17,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+  by encoding them in memory
+  and then writing the encoded result to the file system,
+  and loads images by reading the entire image file into memory
+  and decoding it
+  (rather than letting OpenCV manage the reading/writing of files).
+  This works around OpenCV's lack of support for Unicode paths on Windows
+  (see <https://github.com/opencv/opencv/issues/4292>).
+  (<https://github.com/openvinotoolkit/datumaro/pull/392>)
```
Comment (zhiltsov-max) [inline CHANGELOG.md:28]:
```suggestion
- Saving of images with Unicode paths on Windows (<https://github.com/openvinotoolkit/datumaro/pull/392>)
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -32,6 +41,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Item id in ImageNet format (<https://github.com/openvinotoolkit/datumaro/pull/371>)
 - Double quotes for ICDAR Word Recognition (<https://github.com/openvinotoolkit/datumaro/pull/375>)
 - Wrong display of builtin formats in CLI (<https://github.com/openvinotoolkit/datumaro/issues/332>)
+- The Market-1501 converter now saves the annotation file in UTF-8
+  rather than the locale encoding
+  (<https://github.com/openvinotoolkit/datumaro/pull/392>)
+- The ICDAR, PASCAL VOC and VGGFace2 extractors
+  can now load images from subdirectories on WIndows
+  (<https://github.com/openvinotoolkit/datumaro/pull/392>)
```
Comment (zhiltsov-max) [inline CHANGELOG.md:49]:
```suggestion
- Non utf-8 encoding of annotation files in Market-1501 export (<https://github.com/openvinotoolkit/datumaro/pull/392>)
- Import of ICDAR, PASCAL VOC and VGGFace2 images from subdirectories on WIndows
  (<https://github.com/openvinotoolkit/datumaro/pull/392>)
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -17,6 +17,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Datumaro no longer depends on scikit-image
   (<https://github.com/openvinotoolkit/datumaro/pull/379>)
 - `Dataset` remembers export options on saving / exporting for the first time (<https://github.com/openvinotoolkit/datumaro/pull/386>)
+- Saving of images with Unicode paths on Windows (<https://github.com/openvinotoolkit/datumaro/pull/392>)
```
Comment (zhiltsov-max) [inline CHANGELOG.md:20]:
```suggestion
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -32,6 +33,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Item id in ImageNet format (<https://github.com/openvinotoolkit/datumaro/pull/371>)
 - Double quotes for ICDAR Word Recognition (<https://github.com/openvinotoolkit/datumaro/pull/375>)
 - Wrong display of builtin formats in CLI (<https://github.com/openvinotoolkit/datumaro/issues/332>)
+- Non utf-8 encoding of annotation files in Market-1501 export (<https://github.com/openvinotoolkit/datumaro/pull/392>)
+- Import of ICDAR, PASCAL VOC and VGGFace2 images from subdirectories on WIndows
+  (<https://github.com/openvinotoolkit/datumaro/pull/392>)
```
Comment (zhiltsov-max) [inline CHANGELOG.md:38]:
```suggestion
  (<https://github.com/openvinotoolkit/datumaro/pull/392>)
- Saving of images with Unicode paths on Windows (<https://github.com/openvinotoolkit/datumaro/pull/392>)
```

**Review by zhiltsov-max — APPROVED** [review]

**Pull request merged**